packages/graphql/src/schemaTypes/objectTypes/gql-Migration.ts

```diff
@@ -20,15 +20,15 @@ export const MigrationStep = objectType({
     t.nonNull.boolean('isCurrentStep', {
       description: 'This is the current step',
       resolve: (source, args, ctx) => {
-        return ctx.migration.step === source.name
+        return ctx.coreData.migration.step === source.name
       },
     })
 
     t.nonNull.boolean('isCompleted', {
       description: 'Has the current step been completed',
       resolve: (source, args, ctx) => {
-        const indexOfObservedStep = ctx.migration.filteredSteps.indexOf(source.name)
-        const indexOfCurrentStep = ctx.migration.filteredSteps.indexOf(ctx.migration.step)
+        const indexOfObservedStep = ctx.coreData.migration.filteredSteps.indexOf(source.name)
+        const indexOfCurrentStep = ctx.coreData.migration.filteredSteps.indexOf(ctx.coreData.migration.step)
```

Comment:
Why change that?

We remove a layer of abstraction that IMHO has a very real use in the graphql package.

Reply:
I just moved the data from getting stored in a Source to CoreData. Isn't the idea of a source to derive data from coreData, instead of having multiple places to save things?

The problem I encountered is we need to make sure we fully derive the legacyConfig include the modification from `cypress/plugins/index.js`. This has to happen before we call `initializeFlags`. 

I also found it quite confusing we have data in multiple places - `coreData`, `lifecycleManager` and Sources. We could  have `steps` in migration and derive a subset of `filteredSteps` based on the flags, if that's what you are suggesting here?